subsys/dfu/dfu_target/src/dfu_stream_flatten.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <dfu_stream_flatten.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(dfu_stream_flatten, CONFIG_DFU_TARGET_LOG_LEVEL);
+
+int stream_flash_flatten_page(struct stream_flash_ctx *ctx, off_t off)
+{
+	int rc;
+	struct flash_pages_info page;
+
+#if defined(CONFIG_STREAM_FLASH_ERASE) && defined(CONFIG_FLASH_HAS_EXPLICIT_ERASE) && \
+defined(CONFIG_FLASH_HAS_NO_EXPLICIT_ERASE)
+	bool update_page = false;
+
+	const struct flash_parameters *fparams = flash_get_parameters(ctx->fdev);
+
+	if ((flash_params_get_erase_cap(fparams) & FLASH_ERASE_C_EXPLICIT)) {
+		update_page = true;
+	}
+#endif
+
+	rc = flash_get_page_info_by_offs(ctx->fdev, off, &page);
+
+	if (rc != 0) {
+		LOG_ERR("Error %d while getting page info", rc);
+		return rc;
+	}
+
+#if defined(CONFIG_STREAM_FLASH_ERASE)
+#if defined(CONFIG_FLASH_HAS_EXPLICIT_ERASE) && defined(CONFIG_FLASH_HAS_NO_EXPLICIT_ERASE)
+	if (!update_page) {
+	} else
+#endif
+	if (ctx->last_erased_page_start_offset == page.start_offset) {
+		return 0;
+	}
+#endif
+
+	LOG_DBG("Flattening page at offset 0x%08lx", (long)page.start_offset);
+
+	rc = flash_flatten(ctx->fdev, page.start_offset, page.size);
+
+	if (rc != 0) {
+		LOG_ERR("Error %d while flattening page", rc);
+#if defined(CONFIG_STREAM_FLASH_ERASE)
+#if defined(CONFIG_FLASH_HAS_EXPLICIT_ERASE) && defined(CONFIG_FLASH_HAS_NO_EXPLICIT_ERASE)
+	} else if (!update_page) {
+#endif
+	} else {
+		ctx->last_erased_page_start_offset = page.start_offset;
```

Comment:
if so then decision of updating of this field might be just based one exclusively #if defined(CONFIG_STREAM_FLASH_ERASE).

Reply:
Fixed